AdvLoggerPkg/Library/AdvancedLoggerLib/BaseArm/AdvancedLoggerLib.c

```diff
@@ -49,12 +49,12 @@ ValidateInfoBlock (
     return FALSE;
   }
 
-  if (mLoggerInfo->LogBuffer != (PA_FROM_PTR (mLoggerInfo + 1))) {
+  if (mLoggerInfo->LogBufferOffset != EXPECTED_LOG_BUFFER_OFFSET (mLoggerInfo)) {
     return FALSE;
   }
 
-  if ((mLoggerInfo->LogCurrent > mMaxAddress) ||
-      (mLoggerInfo->LogCurrent < mLoggerInfo->LogBuffer))
+  if (PA_FROM_PTR (LOG_CURRENT_FROM_ALI (mLoggerInfo)) > mMaxAddress) ||

```

Comment:
I do not want to rock the boat too much, but it seems that mMaxAddress will be less useful after switching to offsets? If mMaxAddress is changed to mMaxOffset, we will not need to do calculations here.

Reply:
I completely agree. My plan is to preserve the existing functionality with this initial large PR and file an issue to track cleaning up all of these calculations (there are other things besides mMaxAddress where we use ptr arithmetic instead of just doing it cleanly off of offsets, but I didn't want to rock the boat too much with this PR.

Reply:
Tracked in bug #474